### ISSUE TITLE
HTTP3 QUIC - client over Cronet sends CertificateUnknown and closes QUIC listener

### ISSUE THREAD

**Issue by ValeryYegorov** [dotnet/runtime#79350]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I'm implementing gRPC QUIC for bidirectional client-server interaction. SSL connection creates using letsencrypt certificate. Server side on linux. Client side on Android/IoS. When I connect to gRPC server over linux using WinForms application - all Ok. When I connect using Android/Cronet - client after connection sends 'CertificateUnknown' .

### Expected Behavior

WinForm, Android and IoS client can interact with server using bidirectional streams. If any exception or failure gRPC QUIC listener should continue or restart.

### Steps To Reproduce

On linux server - 'CertificateUnknown'
via TLS - server sends 2 certificates leaf and intermediate.
via QUIC - server sends only 1 leaf certificate without intermediate. 

### .NET Version

.NET SDK 7.0.100

### Anything else?

For additional information please check [grpc-dotnet](https://github.com/grpc/grpc-dotnet/issues/1956)

**Comment by ValeryYegorov** [comment]
Hello @wfurt. 
I prepared certificates as below and added SslStreamCertificateContext.Create - works correctly
```
    certificate = X509Certificate2.CreateFromPemFile("certificate.pem", "private.key");

    chain = new X509Certificate2Collection();
    chain.ImportFromPemFile("ca_bundle.crt");
                
    sslStreamCertificateContext = SslStreamCertificateContext.Create(certificate, chain, true);
```
And init https as below

```
    services.Configure<KestrelServerOptions>(serverOptions =>
            {
               serverOptions.ListenAnyIP(1189, o =>
                    {
                        o.UseConnectionLogging();
                        o.Protocols = HttpProtocols.Http1AndHttp2AndHttp3;
                        o.KestrelServerOptions.Limits.KeepAliveTimeout = TimeSpan.FromSeconds(3600);
                        o.UseHttps(options =>
                        {
                            options.AllowAnyClientCertificate();
                            options.ServerCertificate = certificate;
                            options.ServerCertificateChain = chain;
                        });
                    });

            });
```
Linux server has the same issue. Intermediate certificate missed in QUIC handshake.

**Comment by wfurt** [comment]
That make sense. Could you look at the created `sslStreamCertificateContext` to check if it seems constructed correctly? 
There is private array holding the intermediates so it would be good to verify it is correct. I assume the `ca_bundle.crt` have the intermediates? And is root CA included as well? 

This should generally be covered by 
https://github.com/dotnet/runtime/blob/752cb7e0b7f48f177a0f923164e903f49f2df45f/src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs#L61
On Linux we set the certificates explicitly so there are very few places to look. 
For completeness, did you try to run it on standard Linux distro? I don't know how Amazon Linux would impact this but it may be worth off try.

**Comment by ValeryYegorov** [comment]
Hi @wfurt 
![sslStreamCertificateContext](https://user-images.githubusercontent.com/61559097/206296156-e47211a0-127d-4061-9230-3d2bf0572335.jpg)
That is intermediate certificates from ca_bundle. [0] - intermediate, [1] -root.
>On Linux we set the certificates explicitly so there are very few places to look.

Please explain what you exactly mean.
>For completeness, did you try to run it on standard Linux distro?

No.

**Comment by ValeryYegorov** [comment]
Hi @wfurt 
For test purposes I created new gRPC application from VisualStudio gRPC template. Modifed appsettings by adding port and fullchain certificate
```
"Kestrel": {
    "EndpointDefaults": {
      "Protocols": "Http1AndHttp2AndHttp3"
    },
    "Endpoints": {
      "Https": {
        "Url": "https://*:1191"
      }
    },
    "Certificates": {
      "Default": {
        "Path": "fullchain.pem",
        "KeyPath": "private.key"
      }
    }
  }
```
As cilent-side we use Android/Cronet (same as earlier). 
gRPC started on Windows - all works correctly.
gRPC started on linux (Amazon and Debian)  - same as earlier. On client application start,  server sends only 1 leaf certificate

[netlog_09_12_1.zip](https://github.com/dotnet/runtime/files/10198226/netlog_09_12_1.zip)
[client_log_viewer](https://netlog-viewer.appspot.com/#events)

Please specify what you meant by?
>On Linux we set the certificates explicitly

Best regards.

**Comment by wfurt** [comment]
The logic lives here. https://github.com/dotnet/runtime/blob/c39f2b69e1e8f643b1009088b49dd572506e3f42/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs#L174-L196

You may notice that the "Schannel" route is different. For Linux we create the PCKS12 object and we pass it down to msquic. What happen there is not in our control.  You can verify what is happening there with debugger. 

we may need to collect msquic logs if everything looks ok up to here.
cc: @nibanks

**Comment by wfurt** [comment]
can you post you whole test project @ValeryYegorov or try it with simple QuicListener? My exposure to gRPC was minimal and  we have test that should cover what you describing. So I suspect there is some small difference I'm missing. I can run on Debian version of your choice.

**Comment by ManickaP** [comment]
Triage: we should investigate if there's a product bug and if so fix it for 8.0. We should have this covered by tests though. Leaving in 8.0, marking needs-more-info.

For investigation we need: msquic logs, repro case.
